<!-- -*- mode: xml -*- -->
<launch>
  <arg name="paused" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="vision" default="true"/>
  <arg name="gimbal" default="true"/>
  <arg name="multidrone" default="true"/>
  <arg name="motion" default="true"/>
  <arg name="LOCAL_IP" default="192.168.97.31"/> <!-- UAV -->
  <arg name="REMOTE_IP" default="192.168.97.188"/> <!-- Remote PC -->
  <!-- drone number -->
  <arg name="drone_number" default="1.0"/>
  <arg name="image_topic" default="/image_zenmus"/>
  <arg name="cam_info_topic" default="/dji_sdk/camera_info"/>
  <arg name="raw_image" default="/image_zenmus"/>

  <!-- DJI SDK -->
  <include file="$(find jsk_task3_m100)/launch/jsk_udp.launch" >
         <arg name="LOCAL_IP" value="$(arg LOCAL_IP)"/>
         <arg name="REMOTE_IP" value="$(arg REMOTE_IP)"/>
         <arg name="raw_image" value="$(arg raw_image)" />
  </include>
 <!-- Magnet -->
  <include file="$(find jsk_mbzirc_board)/launch/magnet.launch">
      <arg name="verbose" value="false" />
  </include>

  <group if="$(arg gimbal)">
       <!-- gimbal control, always facing down -->
        <node pkg="jsk_task3_m100" type="gimbal_control" name="gimbal_control">
    </node>
  </group>
       <!-- task3 vision node -->
  <group if="$(arg vision)">
        <node pkg="jsk_task3_m100" type="task3_vision" name="task3_vision">
          <rosparam subst_value="true">
            image_topic: $(arg image_topic)
            cam_info_topic: $(arg cam_info_topic)
          </rosparam>
    </node>
  </group>
       <!-- multidrone  node -->
  <group if="$(arg multidrone)">
        <node pkg="jsk_task3_m100" type="multidrone" name="multidrone">
      <rosparam subst_value="true">
        drone_number: $(arg drone_number)
<!--        origin_latitude: 24.644418
        origin_longitude: 54.755515     -->
        origin_latitude: 24.479555
        origin_longitude: 54.60987
      </rosparam>
    </node>
  </group>

  <group if="$(arg motion)">
       <!-- uav_treasure_pick -->
       <node pkg="jsk_task3_m100" type="uav_treasure_pick" name="uav_treasure_pick">
      <rosparam subst_value="true">
        landing_x: 5.9
        landing_y: 48.5
<!--        landing_x: -31.3921
        landing_y: 39.5614   -->
        drone_number: $(arg drone_number)
      </rosparam>
    </node>
       <!-- uav_treasure_move -->
     <node pkg="jsk_task3_m100" type="uav_treasure_move" name="uav_treasure_move">
    </node> 
  </group>
</launch>
